contracts/airdrop/StakingAirdrop.sol

```diff
@@ -0,0 +1,157 @@
+error ZeroAddress();
+
+/// @dev Provided zero value.
+error ZeroValue();
+
+/// @dev Only `owner` has a privilege, but the `sender` was provided.
+/// @param sender Sender address.
+/// @param owner Required sender address as an owner.
+error OwnerOnly(address sender, address owner);
+
+/// @dev Wrong length of two arrays.
+/// @param numValues1 Number of values in a first array.
+/// @param numValues2 Number of values in a second array.
+error WrongArrayLength(uint256 numValues1, uint256 numValues2);
+
+/// @dev Provided incorrect amount.
+/// @param provided Provided amount.
+/// @param expected Expected amount.
+error WrongAmount(uint256 provided, uint256 expected);
+
+/// @dev Value overflow.
+/// @param provided Overflow value.
+/// @param max Maximum possible value.
+error Overflow(uint256 provided, uint256 max);
+
+/// @dev Caught reentrancy violation.
+error ReentrancyGuard();
+
+
+/// @title StakingAirdrop - Smart contract for staking airdrop
+/// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
+/// @author Andrey Lebedev - <andrey.lebedev@valory.xyz>
+/// @author Mariapia Moscatiello - <mariapia.moscatiello@valory.xyz>
+contract StakingAirdrop {
+    event Claimed(address indexed sender, uint256 indexed serviceId, uint256 amount);
+
+    // Version number
+    string public constant VERSION = "0.1.0";
+    // Airdrop amount
+    uint256 public constant AIRDROP_AMOUNT = 10_000;
+
+    // Token address
+    address public immutable token;
+    // Service registry address
+    address public immutable serviceRegistry;
+
+    // Reentrancy lock
+    uint256 internal _locked = 1;
+
+    // Mapping of service Id => airdrop amount
+    mapping(uint256 => uint256) public mapServiceIdAirdropAmount;
+
+    /// @dev StakingAirdrop constructor.
+    /// @param _token Token address.
+    /// @param _serviceRegistry Service registry address.
+    /// @param _serviceIds Set of service Ids.
+    /// @param _amounts Set of corresponding amounts.
+    constructor(
+        address _token,
+        address _serviceRegistry,
+        uint256[] memory _serviceIds,
+        uint256[] memory _amounts
+    ) {
+        // Check for zero addresses
+        if (_token == address(0) || _serviceRegistry == address(0)) {
+            revert ZeroAddress();
+        }
+
+        // Check array lengths
+        if (_serviceIds.length != _amounts.length) {
+            revert WrongArrayLength(_serviceIds.length, _amounts.length);
+        }
+
+        token = _token;
+        serviceRegistry = _serviceRegistry;
+
+        uint256 checkTotalAmount;
+        // Assign airdrop amounts
+        for (uint256 i = 0; i < _serviceIds.length; ++i) {
+            checkTotalAmount += _amounts[i];
+            mapServiceIdAirdropAmount[_serviceIds[i]] = _amounts[i];
+        }
+
+        if (checkTotalAmount != AIRDROP_AMOUNT) {
+            revert WrongAmount(checkTotalAmount, AIRDROP_AMOUNT);
+        }
+    }
+
+    /// @dev Claims airdrop.
```

Comment:
we need to share with claimers that, in order to claim their airdrop they needs to be unstaked!

Reply:
Refactored no, rewards will go to a service multisig. Also, there is a function claimForAll() that can take care of all (remaining) airdrops

Reply:
Why do they need to be unstaked?